cegs_portal/search/view_models/v1/dna_features.py

```diff
@@ -282,3 +282,17 @@ def target_reo_search(cls, source_id: str):
         )
 
         return reg_effects
+
+    @classmethod
+    def non_targeting_reo_search(cls, gene_pk: int):
+        source_features = DNAFeature.objects.filter(closest_gene_id=gene_pk)
```

Comment:
I think that, instead of calling this once for each feature, you could pass in all the features and just call it once total:

```suggestion
    def non_targeting_reo_search(cls, genes: list[DNAFeature]):
        source_features = DNAFeature.objects.filter(closest_gene__in=genes)
```